csrc/runtime/executor.cpp

```diff
@@ -282,6 +283,45 @@ void KernelExecutor::compile(
   if (isProfilerEnabled()) {
     FusionProfiler::segment(group_id_).stopCompile();
   }
+
+  for (auto expr : exprs) {
+    if (ir_utils::isCpAsyncBulk(expr)) {
+      has_tma_ = true;
+    }
+    if (expr->isA<RNGOp>()) {
+      has_rng_ = true;
+    }
+  }
+
+  // If an output has an alias to an input and is marked Evaluate, then
+  // expression evaluator evaluate is called on that output to produce the meta
+  // data manipulation it requires. If that manipulation is something like a
+  // slice, and that slice has a symbolic integer it depends on, then this
+  // function returns true.
+  //
+  // This could happen for other examples and this function will return true if
+  // to evaluate the output that has an alias, other values besides the aliased
+  // input need to be bound to the expression evaluator to evaluate the output.
+  for (auto output : fusion->outputs()) {
+    if (output->isA<TensorView>()) {
+      auto out_tv = output->as<TensorView>();
+      auto alias_info = fusion->getOutputAlias(out_tv);
+      if (alias_info.type != AllocationType::Evaluate) {
+        continue;
+      }
+      auto aliased_to = alias_info.aliased_io->as<TensorView>();
+      auto inputs = InputsOf::output(out_tv);
+      for (auto input : inputs) {
+        if (input->isA<TensorView>() && input->sameAs(aliased_to)) {
+          continue;
+        }
+      }
+      if (out_tv->isConst()) {
```

Comment:
I think `isConst` on a tensor is always false. What is this part supposed to do?

Reply:
Good catch! This should be input->isConst(), not out_tv->isConst.

Reply:
Fixed